core-metastore/src/main/java/org/zalando/nakadi/repository/db/SchemaRepository.java

```diff
@@ -43,6 +44,22 @@ public EventTypeSchema getSchemaVersion(final String name, final String version)
         }
     }
 
+    public Optional<EventTypeSchema> getLatestSchemaByType(final String name, final EventTypeSchema.Type schemaType) {
+        //
+        // TODO: actually we should order by version, but that is harder to do in the DB
+        //
+        final String sql = "SELECT ets_schema_object FROM zn_data.event_type_schema " +
+                "WHERE ets_event_type_name = ? AND ets_schema_object ->> 'type' = ? " +
+                "ORDER BY ets_schema_object ->> 'created_at' DESC " +
```

Comment:
Btw, `ORDER BY ... created_at` works only as long as we stick to the same time zone and format.  The field itself is treated as string here.

Here and in `SubscriptionDbRepository` we should cast the value to timestamp for ordering (and create the corresponding index to speed it up).